packages/react-ui-kit/src/Form/SelectComponents.tsx

```diff
@@ -134,6 +135,55 @@ export const Menu = (dataUieName: string, css?: CSSObject) => (props: MenuProps)
   );
 };
 
+// eslint-disable-next-line react/display-name
+export const MenuList = (menuListHeading: string, dataUieName: string) => (props: MenuListProps) => {
+  const {selectProps, children} = props;
+
+  const handleClose = () => {
+    if (selectProps && selectProps.onMenuClose) {
+      selectProps.onMenuClose();
+    }
+  };
+
+  return (
+    <components.MenuList {...props}>
+      <div
+        {...(dataUieName && {
+          'data-uie-name': `menu-list-${dataUieName}`,
+        })}
+      >
+        <div
+          css={(theme: Theme) => ({
+            display: 'flex',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+            fontSize: theme.fontSizes.medium,
+            fontWeight: 600,
+            padding: '8px 16px',
+          })}
+        >
+          {menuListHeading}
+          <button
```

Comment:
Consider adding an accessible label such as `aria-label`, without that a screen reader user won't be able to know what this button is for